nums/core/linalg.py

```diff
@@ -256,6 +256,136 @@ def inv(app: ArrayApplication, X: BlockArray):
+        X = X.reshape(block_shape=(X.shape[0]//4, X.shape[0]//4))
+
+    # Setup metadata
+    full_shape = X.shape
+    grid_shape = X.grid.grid_shape
+    block_shape = X.block_shape
+
+    R = X.copy()
+    Zs = app.zeros(full_shape, block_shape, X.dtype)
+
+    # Calculate R_11^-1
+    r11_oid = R.blocks[(0,0)].oid
+    r11_inv_oid = app.cm.inv(r11_oid, syskwargs={
+                                                "grid_entry": (0, 0),
+                                                "grid_shape": grid_shape
+                                            })
+    R.blocks[(0,0)].oid = r11_inv_oid
+    R_tl_shape = block_shape
+
+    # Continue while R_tl.shape != R.shape
+    while R_tl_shape[0] != full_shape[0] and R_tl_shape[1] != full_shape[1]:
+        # Calculate R11
+        R11_block = (
+            int(np.ceil(R_tl_shape[0] // block_shape[0])),
+            int(np.ceil(R_tl_shape[1] // block_shape[1]))
+            )
+        R11_oid = R.blocks[R11_block].oid
+        R11_shape = R.blocks[R11_block].shape
+
+        R11_inv_oid = app.cm.inv(R11_oid, syskwargs={
+                                                        "grid_entry": R11_block,
+                                                        "grid_shape": grid_shape
+                                                    })
+
+        # Reset R11 inplace
+        R.blocks[R11_block].oid = R11_inv_oid
+
+        # Calculate R01
+        R01_oids = []
+        R01_shapes = []
+        R01_grid_entries = []
+        R01_sb_row, R01_sb_col = 0, R11_block[1] # sb -- start_block
+        R01_num_blocks = R11_block[0]
+
+        # Collect data for R01
+        for inc in range(R01_num_blocks):
+            R01_oids.append(R.blocks[(R01_sb_row + inc, R01_sb_col)].oid)
+            R01_shapes.append(R.blocks[(R01_sb_row + inc, R01_sb_col)].shape)
+            R01_grid_entries.append((R01_sb_row + inc, R01_sb_col))
+
+        # Perform matrix multiplication: R01_1 = -R00 @ R01
+        R01_1_oids = []
+        for row_block in range(R01_num_blocks):
+            sub_oids = []
+
+            for col_block in range(R01_num_blocks):
+
+                # Get data for R00
+                R00_oid = R.blocks[(row_block, col_block)].oid
+                Z_oid = Zs.blocks[(row_block, col_block)].oid
+
+                R00_bs = R.blocks[(row_block, col_block)].shape
+
+                # Calculate -R00 = 0 - R00
+                neg_R00_oid = app.cm.bop("subtract", Z_oid, R00_oid,
+                                                False, False, axes=1, syskwargs={
+                                                    "grid_entry": (row_block, col_block),
+                                                    "grid_shape": grid_shape
+                                                })
+                # Calculate -R00 @ R01
+                sub_oids.append(app.cm.bop("tensordot", neg_R00_oid, R01_oids[col_block],
+                                    False, False, axes=1, syskwargs={
```

Comment:
FYI @elibol @dlzou Thanks for the heads-up on the API change! I tested the test locally and it passes, so hopefully the build will pass as well.

Reply:
Linting errors should be fixed now as well

Reply:
ugh... linting is a pain